### ISSUE TITLE
Editing a photo loses EXIF

### ISSUE THREAD

**Issue by semberal** [ente-io/ente#576]
Hi, when I edit a photo (using the pencil icon) and save copy of the photo back to ente, EXIF information is lost.

**Comment by StefanOltmann** [comment]
To rotate or mirror a JPG, you only need to update the TIFF orientation tag in the EXIF metadata. This avoids the need to re-save the entire file, preventing any loss of quality or other metadata.

**Comment by edisso999** [comment]
Any news on this? Editing an image and losing all Exif data is a no go.

**Comment by StefanOltmann** [comment]
@edisso999 Agreed.

I recommend using a proper photo organizer like [Ashampoo Photo Organizer](https://ashampoo.com/photo-organizer).
Ente is just a photo backup app; you shouldn't work on your files there.

**Comment by edisso999** [comment]
Ente is not 'just a photo backup app'. And it offers edits - so it should do it right.

BTW: Considering something from Ashampoo as 'proper' is ...

**Comment by StefanOltmann** [comment]
Yeah, it „should“, but obviously it doesn’t - and that seems not to change anytime soon. Without proper metadata handling I can’t use the edit functionality - that’s the point. So it’s in fact eventually just a photo backup app for me. Therefore I do my metadata handling and edits with other tools - proper tools for the job.

**Comment by jmachan** [comment]
I'm upvoting strongly. I was happy to find an alternative to google and de-google my whole photo archive, but I used to make basic edits in google photos. The fact I can't do it in Ente without loosing EXIF is painful :(

**Comment by gszasz** [comment]
It does not have to preserve the EXIF data universally, but it should with non-destructive edits like rotating 90°. Most people expect behavior similar to Google Photos. Lack of this functionality prevents me from migrating my Google Photos archive to Ente.

**Comment by kutzi** [comment]
There should be a warning at least IMO. I've now edited (rotated) several photos already and deleted the original - not being aware that EXIF info was lost.
Any destructive operation should come with a warning!

**Comment by gszasz** [comment]
The recent update finally preserves the original timestamps of the edited images but still destroys original EXIF when rotating an image. Correction of the image orientation is one of the most used post-processing corrections, if not _the_ most used correction. In 2026, this should even be corrected automatically regardless whether this would be completely non-destructive or altering the "Orientation" EXIF.

**Comment by StefanOltmann** [comment]
They just don’t care. I stopped using Ente at all a while ago.

**Comment by zilexa** [comment]
The changelog (only for the Desktop app) says:
> metadata preservation on editing

https://ente.com/help/photos/changelog

If what @gszasz says is true:

> The recent update finally preserves the original timestamps of the edited images but still destroys original EXIF when rotating an image. Correction of the image orientation is one of the most used post-processing corrections, if not _the_ most used correction. In 2026, this should even be corrected automatically regardless whether this would be completely non-destructive or altering the "Orientation" EXIF.

Then the changelog is quite misleading.. 

Its hard to win back trust like this.

**Comment by gszasz** [comment]
I'm still not using Ente. It does not really make sense to migrate my Google Photos to Ente if it is incapable of fixing the orientation of my pictures. If they keep ignoring this, I will most likely delete my Ente account and migrate my Google Photos to NextCloud instead. @zilexa was right, the changelog is misleading. Even though the basic metadata (like date and time) is indeed preserved, the EXIF metadata is lost by editing.

**Comment by edisso999** [comment]
Will this ever be changed/corrected?

**Comment by zilexa** [comment]
This is a major issue in my opinion. For several reasons including misleading paid subscribers that might not discover the loss of metadata until years later. 

This issue and https://github.com/ente/ente/discussions/543 are the ones that will make me not renew my subscription early next year. 

There are other quirks, like with shared albums, when a non-owner uploads photos to it. They are not stored in their account. I can see them but don't have them. You have to also send them an upload link (as if they are a non Ente user) to get all the files in 1 album.

**Comment by ashil-pilot** [comment]
An update on the mobile side of this issue: starting with Photos v1.3.63, edited copies retain key camera, capture-date, and location EXIF fields. For JPEG images where rotation or flipping is the only edit, the mobile editor uses a lossless path: the original JPEG pixels are kept and only the EXIF orientation is changed.

Relevant mobile PRs:
- https://github.com/ente/ente/pull/11177 — preserves EXIF information in edited copies
- https://github.com/ente/ente/pull/11337 — adds lossless JPEG rotation/flip

There are still scope limitations on mobile: re-rendered edits copy a selected set of important EXIF fields rather than every possible/vendor-specific tag, and lossless rotation/flip currently applies only to JPEGs. Non-JPEG images and edits combined with crop, filters, tuning, text, etc. are exported through the rendered JPEG path.

This update covers the mobile editor only; it does not confirm that the original web/desktop editing path reported here is fixed.

**Comment by zilexa** [comment]
Wow this really really good news and a clear sign of users being heard and things like this getting prioritised, thank you!!
